test/Microsoft.VisualStudio.Editor.Razor.Test/Infrastructure/TestTextBuffer.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Utilities;
+
+namespace Microsoft.VisualStudio.Test
+{
+    public class TestTextBuffer : ITextBuffer
```

Comment:
Did you ask about those mocks?

Reply:
Talked to the WTE folks and they rely on a privately constructed MEF service catalog (probing paths etc.) to create buffer factories; insufficient for our usecase. Will need to reach out to others.